ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto

```diff
@@ -111,14 +116,25 @@ message GetTransactionByIdRequest {
   // Required
   repeated string requesting_parties = 3;
 
+  // This flag indicates whether the response should contain a TransactionTree (in case of false)
+  // or the flat Transaction (in case of true).
+  // Optional
+  bool return_flat_transaction = 4;
+
   // Server side tracing will be registered as a child of the submitted context.
   // This field is a future extension point and is currently not supported.
   // Optional
   TraceContext trace_context = 1000;
 }
 
 message GetTransactionResponse {
-  TransactionTree transaction = 1;
+  // The transaction tree in case of return_flat_transaction == false
+  // Optional
+  TransactionTree transaction_tree = 1;
+
+  // The transaction tree in case of return_flat_transaction == true
+  // Optional
+  Transaction flat_transaction = 2;
```

Comment:
I'm not terribly excited about using the term "flat transaction", but I also want to avoid just changing the type for the field name `transaction` from `TransactionTree` to `Transaction`.
Notice: renaming a field doesn't break wire compatibility.